src/functional/program_processors/codegens/gtfn/codegen.py

```diff
@@ -124,7 +124,12 @@ def visit_OffsetLiteral(self, node: gtfn_ir.OffsetLiteral, **kwargs: Any) -> str
     )
 
     def visit_FunCall(self, node: gtfn_ir.FunCall, **kwargs):
-        return self.generic_visit(node, fun_name=self.visit(node.fun))
+        if isinstance(node.fun, gtfn_ir.SymRef) and node.fun.id in self.user_defined_function_ids:
+            fun_name = f"{self.visit(node.fun)}{{}}()"
```

Comment:
User-defined functions are structs with call operator returning the actual function, i.e. `{}` for creating the instance `()` for the call operator, see below.